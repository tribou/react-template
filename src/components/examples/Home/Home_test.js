// @flow

// Enzyme docs:
// http://airbnb.io/enzyme/docs/api/index.html

import React from 'react'
import { shallow } from 'enzyme'
import Home from './Home'


type OverrideProps = {
  logout: Function,
  history: Object,
}

const mockProps: OverrideProps = {
  logout: () => {},
  history: {
    replace: () => {},
  },
}

it('renders the Home page content', () => {

  const wrapper = shallow(
    <Home
      {...mockProps}
    />
  )

  expect(wrapper).toMatchSnapshot()

})
