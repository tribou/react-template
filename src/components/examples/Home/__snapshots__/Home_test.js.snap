// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Home page content 1`] = `
<div
  className="home"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Home"
  />
  This is the app...
  <ul>
    <li>
      This is a test item
    </li>
  </ul>
  <Link
    replace={false}
    to="/profile"
  >
    <div
      className="logo link"
    />
  </Link>
  <Link
    replace={false}
    to="/todos"
  >
    <div
      className="facebookIcon link"
    />
  </Link>
  <withRouter(Connect(RequireAuth))>
    <button
      onClick={[Function]}
    >
      Logout
    </button>
  </withRouter(Connect(RequireAuth))>
</div>
`;
